skilmap01.py

```diff
@@ -0,0 +1,120 @@
+
+    def convertListToString(values: list[int]):
+        convertedList = [str(item) for item in values]
+        return (",").join(convertedList)
+
+    convertedValues = convertListToString(values)
+
+    return convertedValues == convertListToString(ascList) or convertedValues == convertListToString(descList)
+
+# 課題2
+# 2進数を10進数に変換する関数を作成してください。
+# 例えば、引数に2進数1011が渡されたら10進数11を返します。
+def convertDecimalNumber(value):
+    values = list(str(value))
+    values.reverse()
+    result = 0
+
+    for index, item in enumerate(values):
+        result =+ result + (int(item) * (2 ** index))
+
+    return result
+
+# 課題3
+# 引数に整数nが与えられた場合、1からnまでの各数値に対して、「Loves me」と「Loves me not」という交互に出力した文字列を返す関数を作成してください。
+# 例えば、引数に5が渡された場合は“Loves me, Loves me not, Loves me, Loves me not, Loves me”を返します。
+def outputMessage(value):
+    list = range(value)
+    result = []
+
+    for item in list:
+        if item % 2 == 0:
+            result.append("Loves me")
+        else:
+            result.append("Loves me not")
+
+    return (", ").join(result)
+
+# 課題4
+# トリボナッチ数列のn番目の数値を返す関数を作成してください。
+# トリボナッチ数列はフィボナッチ数列をアレンジしたもので、各数字は前の3つの数字の合計になります。例：0, 1, 1, 2, 4, 7・・・
+# 例えば、引数に0を渡した場合は0、5を渡した場合は7を返します。
+def getTribonacciNumber(n):
+    firstSequence = [0, 0, 1]
+
+    if n < len(firstSequence):
+        return firstSequence[n]
+
+    a, b, c = firstSequence
+
+    for _ in range(3, n + 1):
+        currentValue = a + b + c
+        a, b, c = b, c, currentValue
+
+    return c
+
+# 課題5
+# クレジットカード番号を受け取り、最後の4桁を除くすべての数字をアスタリスクに置き換えた文字列を返す関数を作成してください。
+# 例えば、引数に整数1234567812345678を渡した場合は文字列************5678を返します。
+def convertHideCreditCardNumber(value):
+    convertedStringCardNumber = str(value)
+    visibleCardNumber = convertedStringCardNumber[-4:]
+    hideNumberLength = len(convertedStringCardNumber) - 4
+
+    return ("*" * hideNumberLength) + visibleCardNumber
+
+# 課題6
+# 引数で渡された文字列をスポンジケースに変換する関数を作成してください。
+# スポンジケースとは、文字が小文字と大文字で交互に表示されるスタイルです。
+# 例えば、引数に文字列spongeCaseを渡した場合は文字列sPoNgEcAsEを返します。
+def convertSpongeCase(value):
+    values = list(value)
+    result = ""
+
+    for index, item in enumerate(values):
+        if index % 2 == 0:
+            result = result + item.lower()
+        else:
+            result = result + item.upper()
+
+    return result
+
+# 課題7
+# エンコードする文字列と、各文字をシフトする位置の数である整数の2つの引数を受け取り、文字列を特定の位置数シフトして暗号化する関数を作成してください。
+# 例えば、引数に文字列Hello, Worldと整数3を渡した場合は文字列Khoor, Zruogを返します。
+def encodeValue(value, shiftNumber):
+    splittedValues = list(value)
+    result = ""
+
+    for item in splittedValues:
+        unicodePoint = ord(item)
+        isString = str(item).isalpha() or str(item).isascii()
+
+        if isString:
+            result = result + chr(unicodePoint + shiftNumber)
+        else:
+            result = result + item
+
+    return result
+
+# 課題8
+# 整数のリストを受け取り、それらの整数の積がそれらの整数の合計で割り切れるかどうかを返す関数を作成してください。
+# 例えば、割り切れる場合はTrueを返し、割り切れない場合はFalseを返します。
+def checkDivisibleValue(values):
+    result = False
+    sum = 0
+    product = 0
```

Comment:
組み込み関数と標準ライブラリを使うこともできます

```suggestion
    total_sum = sum(values)
    total_product = math.prod(values)
```